### PR TITLE
Fix incorrect zh-cn translation for Look Up Label in selection controls

### LINKED ISSUE #141764: Incorrect Chinese localization in iOS text selection menu
### Steps to reproduce

In a Flutter app, select a piece of text in a `SelectableText` on iOS

### Expected results

IMG_0051

"Look up" in this case should translate to "查询" (image above shows native iOS translation), which mean "Looking up something (like a term)"

### Actual results

截屏 2024-01-18 18 01 59

However, in a Flutter app, "Look up" is translated to "往上看", which means "Lifting your head and looking upwards", like "Look up to the sky" that kind of look up. It's kind of a funny mistake.

### Code sample

N/A

### Screenshots or Video

N/A

### Logs

N/A

### Flutter Doctor output

```
[✓] Flutter (Channel beta, 3.19.0-0.1.pre, on macOS 14.2 23C64 darwin-arm64, locale en-US)
    • Flutter version 3.19.0-0.1.pre on channel beta at /Users/david/fvm/versions/beta
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1a9a60d044 (8 days ago), 2024-01-10 13:38:09 -0800
    • Engine revision bbebee1465
    • Dart version 3.3.0 (build 3.3.0-279.0.dev)
    • DevTools version 2.31.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0-rc1)
    • Android SDK at /Users/david/Library/Android/sdk
    • Platform android-33, build-tools 34.0.0-rc1
    • ANDROID_HOME = /Users/david/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin version 223.8888

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • ips (mobile)    • 00008112-001424D10AFB401E • ios            • iOS 17.2 21C62
    • macOS (desktop) • macos                     • darwin-arm64   • macOS 14.2 23C64 darwin-arm64
    • Chrome (web)    • chrome                    • web-javascript • Google Chrome 120.0.6099.234
    ! Error: Browsing on the local area network for wiphone (2). Ensure the device is unlocked and attached with a cable or associated with the same local area
      network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### PR DESCRIPTION
**Pull request by LouiseHsu**
Fixes https://github.com/flutter/flutter/issues/141764

Translation suggestion here:
https://tc.corp.google.com/btviewer/edittranslation?project=Flutter&msgId=8222331119728136330&language=zh-CN

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by HansMuller** [comment]
LGTM

**Review by HansMuller — APPROVED** [review]
LGTM. 

This change will eventually be overwritten by Google's translation service; hopefully there will be no effective change when that happens.

File: packages/flutter_localizations/lib/src/l10n/cupertino_zh.arb [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "searchTextFieldPlaceholderLabel": "搜索",
   "noSpellCheckReplacementsLabel": "找不到替换文字",
   "menuDismissLabel": "关闭菜单",
-  "lookUpButtonLabel": "向上看",
+  "lookUpButtonLabel": "查询",
```
Comment (hellohuanlin) [inline packages/flutter_localizations/lib/src/l10n/cupertino_zh.arb:27]: do we have a separate localization for traditional chinese?

**Review by justinmc — APPROVED** [review]
LGTM :+1:

File: packages/flutter_localizations/test/material/translations_test.dart [hunk 2]
```diff
@@ -549,4 +549,13 @@ void main() {
...
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/141764
+  testWidgets('zh-CN translation for look up label', (WidgetTester tester) async {
+    const Locale locale = Locale('zh');
+    expect(GlobalCupertinoLocalizations.delegate.isSupported(locale), isTrue);
+    final MaterialLocalizations localizations = await GlobalMaterialLocalizations.delegate.load(locale);
+    expect(localizations, isA<MaterialLocalizationZh>());
+    expect(localizations.lookUpButtonLabel, '查询');
+  });
```
[the remark this answers is not in the document]
Comment (LouiseHsu) [reply packages/flutter_localizations/test/material/translations_test.dart:560]: added it just cuz the translations havent been approved yet, so small change this change might get overwritten by the translation tool

**Pull request merged**